source/frontend/src/components/propertySelector/search/PropertySelectorSearchContainer.tsx

```diff
@@ -87,17 +91,28 @@ export const PropertySelectorSearchContainer: React.FC<IPropertySelectorSearchCo
           );
         });
 
+        const getPimsTasks = result.features.map(p => {
+          return loadPimsProperties({
+            PID: p?.properties?.PID_NUMBER?.toString(),
+            PIN: p?.properties?.PIN?.toString(),
+          });
+        });
+
         const addressResults = await Promise.all(getAddressTasks);
         const regionDistrictResults = await Promise.all(matchTasks);
+        const pimsResults = await Promise.all(getPimsTasks);
+
         const locations = result.features.map((p, i) => {
           const foundProperty = featureToLocationFeatureDataset(p);
           foundProperty.regionFeature = regionDistrictResults[i]?.regionFeature;
           foundProperty.districtFeature = regionDistrictResults[i]?.districtFeature;
-          foundProperty.pimsFeature = {
-            properties: {
-              STREET_ADDRESS_1: addressResults[i]?.fullAddress,
-            },
-          } as Feature<Geometry, PIMS_Property_Location_View>;
+          foundProperty.pimsFeature = pimsResults[i]?.features?.length
+            ? pimsResults[i]?.features[0]
+            : ({
+                properties: {
+                  STREET_ADDRESS_1: addressResults[i]?.fullAddress,
+                },
+              } as Feature<Geometry, PIMS_Property_Location_View>);
```

Comment:
This casting is smelly

Reply:
This isn't one of my changes, so I'll leave it as-is.